### PR TITLE
don't validate endpoint config with --secret

### PR DESCRIPTION
**Pull request by fauna-chase**
ENG-5661
This came up while I was testing updating schema in a pipeline.  In this scenario the endpoints present in the project configuration won't exist, which is ok as long as a secret is getting passed.

### CODE REVIEW

File: src/lib/config/index.ts [hunk 1]
```diff
@@ -250,7 +248,7 @@ export class ShellConfig {
       );
     }
 
-    if (endpointName === undefined) {
+    if (endpointName === undefined || secretFlag !== undefined) {
```
[the remark this answers is not in the document]
Comment (fauna-chase) [reply src/lib/config/index.ts:251]: I think this is a problem for our CD use case, there will be an endpoint set from the configuration file which sets the endpointName.  so I guess what I'm looking for here is secret is set and the absence of a root config.

[the remark this answers is not in the document]
Comment (macmv) [reply src/lib/config/index.ts:251]: right. so this change looks good, just the below part that says `url: urlFlag` should instead be `url: urlFlag ?? endpoint.url` or something.

[the remark this answers is not in the document]
Comment (fauna-chase) [reply src/lib/config/index.ts]: yeah that makes sense, but really I am targeting an empty root config scenario here, at least that was my intent, so I updated the check to account for that, lmk what you think

**Review by macmv — APPROVED** [review]

File: src/lib/config/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (macmv) [reply src/lib/config/index.ts]: looks good to me

**Pull request merged**